### PR TITLE
Bind consumer with particular shard of Kinesis stream

### PR DESCRIPTION
**Pull request by dmytrodanilenkov**
https://github.com/spring-cloud/spring-cloud-stream-binder-aws-kinesis/issues/35

### CODE REVIEW

**Review by artembilan — CHANGES_REQUESTED** [review]
LGTM, but, please, add a new property to the documentation: https://github.com/spring-cloud/spring-cloud-stream-binder-aws-kinesis/blob/main/spring-cloud-stream-binder-kinesis-docs/src/main/asciidoc/overview.adoc#kinesis-consumer-properties
With respective restriction note about KCL and `instanceCount`.

Thank you!

**Comment by dmytrodanilenkov** [comment]
@artembilan added docs. Also added missing javadoc to pass checkstyle violation(maybe you have better idea how to describe this constant)

File: spring-cloud-stream-binder-kinesis-docs/src/main/asciidoc/overview.adoc [hunk 1]
```diff
@@ -274,6 +274,15 @@ The stream and consumer group (plus workerId) must be provided in the respective
 When `KclMessageDrivenChannelAdapter` endpoint is configured in the binder, it selects an appropriate `KinesisClientLibConfiguration` from the application context according a destination (stream) for binding.
 If there is no `KinesisClientLibConfiguration` bean for its stream, the `KclMessageDrivenChannelAdapter` falls back to original configuration with defaults options for its internal `KinesisClientLibConfiguration`.
 
+shardId::
+The `String` value indicating Kinesis consumer channel adapter should use specific shardId for consuming messages.
+For example: `shardId:shardId-000000000001`.
++
+No default - means no specific shardId selected.
+
+NOTE: KCL doesn't expose any option to deal with a specific shard. When `shardId` property is used, it won't work with KCL. Also, in case
```
Comment (artembilan) [inline spring-cloud-stream-binder-kinesis-docs/src/main/asciidoc/overview.adoc:283]: We prefer *One sentence per line*: https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line

Also I wouldn't say `KCL` in the official doc. Better to call product its name: `Kinesis Client Library`.

You have some other text issue, but I'm going to fix them myself on merge.

**Review by artembilan — APPROVED** [review]
LGTM.

I'm going to pull it locally for final clean up and merge.

Will let you know when it reaches the `upstream/main`!

P.S. It is better to create a top-level branch for the particular fix. Don't PR from the `main`. You are going to have some headache when I merge and you'd like to pull upstream changes locally. `git reset hard` - or something like that you are gong to be needed.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-cloud/spring-cloud-stream-binder-aws-kinesis/commit/a3311e1fbd1dd0de310363e6615575554e132f46.

@dmytrodanilenkov ,

thank you very much for contribution; looking forward for more!

File: spring-cloud-stream-binder-kinesis-docs/src/main/asciidoc/overview.adoc (hunk 1 above)
Comment (dmytrodanilenkov) [reply spring-cloud-stream-binder-kinesis-docs/src/main/asciidoc/overview.adoc:283]: Thank you for detail answer!

**Pull request closed without merging**